dali/operators/generic/one_hot.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2017-2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_ONE_HOT_H_
+#define DALI_OPERATORS_GENERIC_ONE_HOT_H_
+
+#include <vector>
+
+#include "dali/pipeline/operator/operator.h"
+#include "dali/kernels/kernel_params.h"
+#include "dali/core/tensor_view.h"
+#include "dali/core/static_switch.h"
+
+#define ONE_HOT_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float, double)  // NOLINT
+
+namespace dali {
+
+namespace detail {
+template <typename Out, typename In, int ndims = 1>
```

Comment:
```suggestion
template <typename Out, typename In>
```
This function is not actualy templated against ndims.
For generic case the output should have ndims+1 - typical case has 0D (scalar) input - unfortunately, we don't have first class support for 0D tensors.

Reply:
changed